test/testutil/createuvm.go

```diff
@@ -0,0 +1,87 @@
+package testutil
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Microsoft/hcsshim/internal/uvm"
+	"github.com/containerd/containerd"
+)
+
+// CreateWCOWUVM creates a WCOW utility VM with all default options. Returns the
+// UtilityVM object; folder used as its scratch
+func CreateWCOWUVM(ctx context.Context, t *testing.T, client *containerd.Client, id, image string) (*uvm.UtilityVM, []string, string) {
+	return CreateWCOWUVMFromOptsWithImage(ctx, t, client, uvm.NewDefaultOptionsWCOW(id, ""), image)
+
+}
+
+// CreateWCOWUVMFromOptsWithImage creates a WCOW utility VM with the passed opts
+// builds the LayerFolders based on `image`. Returns the UtilityVM object;
+// folder used as its scratch
+func CreateWCOWUVMFromOptsWithImage(ctx context.Context, t *testing.T, client *containerd.Client, opts *uvm.OptionsWCOW, image string) (*uvm.UtilityVM, []string, string) {
+	if opts == nil {
+		t.Fatal("opts must be set")
+	}
+
+	layers := LayerFolders(ctx, t, client, image)
+	scratch := t.TempDir()
+	opts.LayerFolders = append(opts.LayerFolders, layers...)
+	opts.LayerFolders = append(opts.LayerFolders, scratch)
+
+	return CreateWCOWUVMFromOpts(ctx, t, client, opts), layers, scratch
+}
+
+// CreateWCOWUVMFromOpts creates a WCOW utility VM with the passed opts.
+func CreateWCOWUVMFromOpts(ctx context.Context, t *testing.T, _ *containerd.Client, opts *uvm.OptionsWCOW) *uvm.UtilityVM {
```

Comment:
why do we pass through the client here if it's not used?

Reply:
Basically `CreateWCOWUVMFromOptsWithImage` needs it, so I added it to all the functions here to maintain signature consistency, but I can remove it everywhere except for the two functions that need it.